### PR TITLE
Docs: Fix documentation typos in C files

### PR DESCRIPTION
**Pull request by ember91**
That is, documentation only typos in .c and .h files.

See https://github.com/python/cpython/pull/125473 for context.

### CODE REVIEW

File: Include/internal/pycore_critical_section.h [hunk 1]
```diff
@@ -53,13 +53,13 @@ extern "C" {
 
 // Asserts that the mutex is locked.  The mutex must be held by the
 // top-most critical section otherwise there's the possibility
-// that the mutex would be swalled out in some code paths.
+// that the mutex would be stalled out in some code paths.
```
Comment (ember91) [inline Include/internal/pycore_critical_section.h:56]: Not sure what `swalled` is supposed to mean. I can just revert this change and the same one below if you want.

**Comment by ember91** [comment]
> Most changes to Python [require a NEWS entry](https://devguide.python.org/core-developers/committing/#updating-news-and-what-s-new-in-python). Add one using the [blurb_it](https://blurb-it.herokuapp.com) web app or the [blurb](https://pypi.org/project/blurb/) command-line tool.
> 
> If this change has little impact on Python users, wait for a maintainer to apply the `skip news` label instead.

Since this commit/PR only includes documentation only changes, a NEWS addition should not be necessary.

File: Modules/posixmodule.c [hunk 2]
```diff
@@ -5716,7 +5716,7 @@ os_nice_impl(PyObject *module, int increment)
 
     /* There are two flavours of 'nice': one that returns the new
        priority (as required by almost all standards out there) and the
-       Linux/FreeBSD one, which returns '0' on success and advices
+       Linux/FreeBSD one, which returns '0' on success and advises
```
Comment (erlend-aasland) [inline Modules/posixmodule.c:5719]: Please revert. `advices` is perfectly legal.

Comment (ember91) [reply Modules/posixmodule.c:5719]: Advices seems to be plural of the noun while advises is the verb. I think this is right.

File: Modules/_testcapi/exceptions.c [hunk 3]
```diff
@@ -361,7 +361,7 @@ _testcapi_unstable_exc_prep_reraise_star_impl(PyObject *module,
 
 
 /*
- * Define the PyRecurdingInfinitelyError_Type
+ * Define the PyRecurringInfinitelyError_Type
```
Comment (vadmium) [inline Modules/_testcapi/exceptions.c:364]:
```suggestion
 * Define the PyRecursingInfinitelyError type
```

File: Include/internal/pycore_critical_section.h (hunk 1 above)
Comment (vadmium) [reply Include/internal/pycore_critical_section.h:56]: @DinoV wrote this.

Refuse the temptation to guess unless you know what it means.

Having said that, my best guess is “swapped out”. The LL and PP keys are in the same corner of my keyboard.

**Comment by ember91** [comment]
Closing this PR. I will follow the suggestions by @willingc in https://github.com/python/cpython/pull/125569#pullrequestreview-2371548312, but also its follow up https://github.com/python/cpython/pull/125569#issuecomment-2416174012.

I will apply all suggestions in this PR this far in new PRs, obviously.

**Pull request closed without merging**